poseidon/src/lib.rs

```diff
@@ -0,0 +1,43 @@
+mod parameters;
+
+type CircuitField = ark_bls12_381::Fr;
+type FpVar = ark_r1cs_std::fields::fp::FpVar<CircuitField>;
+
+pub mod hash {
+    use ark_bls12_381::Fr;
+    use poseidon_permutation::Instance;
+
+    use crate::parameters::RATE_1_PARAMETERS;
+    /// hashes one field value, outputs a fixed length field value
+    pub fn one_to_one_hash(domain_separator: &Fr, value: Fr) -> Fr {
```

Comment:
what is `domain_separator`?

Reply:
It's arbitrary and used for encoding use cases (so nothing security critical). Basically because you can have several different "poseidons" (variable length, fixed length, various hashing rates etc) you will use this to encode them. 

There is a section explaining it [here](https://spec.filecoin.io/#section-algorithms.crypto.poseidon.domain-separation) and in the original Poseidon paper.